drivers/ieee802154/ieee802154_rf2xx_iface.h

```diff
@@ -0,0 +1,153 @@
+ */
+void rf2xx_iface_phy_tx_start(struct device *dev);
+
+/**
+ * @brief Reads current value from a transceiver register
+ *
+ * This function reads the current value from a transceiver register.
+ *
+ * @param[in] dev   Transceiver device instance
+ * @param[in] addr  Specifies the address of the trx register
+ * from which the data shall be read
+ *
+ * @return value of the register read
+ */
+u8_t rf2xx_iface_reg_read(struct device *dev, u8_t addr);
+
+/**
+ * @brief Writes data into a transceiver register
+ *
+ * This function writes a value into transceiver register.
+ *
+ * @param[in] dev   Transceiver device instance
+ * @param[in] addr  Address of the trx register
+ * @param[in] data  Data to be written to trx register
+ *
+ */
+void rf2xx_iface_reg_write(struct device *dev, u8_t addr, u8_t data);
+
+/**
+ * @brief Subregister read
+ *
+ * @param[in] dev   Transceiver device instance
+ * @param[in] addr  offset of the register
+ * @param[in] mask  bit mask of the subregister
+ * @param[in] pos   bit position of the subregister
+ *
+ * @return  value of the read bit(s)
+ */
+u8_t rf2xx_iface_bit_read(struct device *dev, u8_t addr,
+	u8_t mask, u8_t pos);
+
+/**
+ * @brief Subregister write
+ *
+ * @param[in]   dev       Transceiver device instance
+ * @param[in]   reg_addr  Offset of the register
+ * @param[in]   mask      Bit mask of the subregister
+ * @param[in]   pos       Bit position of the subregister
+ * @param[out]  new_value Data, which is muxed into the register
+ */
+void rf2xx_iface_bit_write(struct device *dev, u8_t reg_addr,
+	u8_t mask, u8_t pos, u8_t new_value);
+
+/**
+ * @brief Reads frame buffer PHR information
+ *
+ * This function reads PHR from the frame buffer of the transceiver.
+ *
+ * @param[in]   dev     Transceiver device instance
+ * @param[out]  phr     Pointer to the Phy Header
+ * buffer.
+ */
+void rf2xx_iface_frame_phr(struct device *dev, u8_t *phr);
+
+/**
+ * @brief Reads frame buffer of the transceiver
+ *
+ * This function reads the frame buffer of the transceiver.
+ *
+ * @param[in]   dev     Transceiver device instance
+ * @param[out]  data    Pointer to the location to store frame
+ * @param[in]   length  Number of bytes to be read from the frame
+ * @param[out]  lqi     Pointer to the Link Quality Indication (LQI)
+ * @param[out]  ed      Pointer to the Energy Detection (ED)
+ * @param[out]  trac    Pointer to the Transaction Status
+ * buffer.
+ */
+void rf2xx_iface_frame_read(struct device *dev, u8_t *data,
+	u8_t length, u8_t *lqi, u8_t *ed, rf2xx_trx_state_trac_t *trac);
+/**
+ * @brief Writes data into frame buffer of the transceiver
+ *
+ * This function writes data into the frame buffer of the transceiver
+ *
+ * @param[in] dev    Transceiver device instance
+ * @param[in] data   Pointer to data to be written into frame buffer
+ * @param[in] length Number of bytes to be written into frame buffer
+ */
+void rf2xx_iface_frame_write(struct device *dev, u8_t *data, u8_t length);
+
+/**
```

Comment:
about the 3 functions below, they are not used in the driver (in this commit at least). Remove them then (and insert them again in the commit that actually do use it)

Reply:
Agree! We need improve Zephyr driver iface to get access to HW features.